tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Verify/VerifySetupTool.cs

```diff
@@ -0,0 +1,326 @@
+    }
+
+    private async Task<List<SetupRequirements.Requirement>> GetCoreRequirements(CancellationToken ct)
+    {
+        var assembly = Assembly.GetExecutingAssembly();
+        var names = assembly.GetManifestResourceNames();
+        using var stream = assembly.GetManifestResourceStream("Azure.Sdk.Tools.Cli.Configuration.RequirementsV1.json");
+        using var reader = new StreamReader(stream);
+        var requirementsJson = await reader.ReadToEndAsync(ct);
+
+        var setupRequirements = JsonSerializer.Deserialize<SetupRequirements>(requirementsJson);
+
+        if (setupRequirements == null)
+        {
+            throw new Exception("Failed to parse requirements JSON.");
+        }
+
+        var reqs = new List<SetupRequirements.Requirement>();
+        foreach (var kv in setupRequirements.categories)
+        {
+            var category = kv.Key;
+            var requirements = kv.Value;
+            if (string.Equals(category, "core", StringComparison.OrdinalIgnoreCase))
+            {
+                if (requirements != null)
+                {
+                    reqs.AddRange(requirements);
+                }
+            }
+        }
+
+        return reqs;
+    }
+
+    private string CheckVersion(string output, SetupRequirements.Requirement req)
+    {
+        // Return empty string if version requirement is satisfied, else return version to upgrade to
+        string versionPattern = @"(>=|<=|>|<|=)\s*([\d\.]+)";
+        
+        var match = System.Text.RegularExpressions.Regex.Match(req.requirement, versionPattern);
+        if (!match.Success)
+        {
+            // No version specified in the requirement
+            return String.Empty;
+        }
+
+        string operatorSymbol = match.Groups[1].Value;
+        string requiredVersion = match.Groups[2].Value;
+
+        logger.LogInformation("Requires version: {requiredVersion}", requiredVersion);
+
+        // Parse the output version
+        var outputVersionMatch = System.Text.RegularExpressions.Regex.Match(output, @"[\d\.]+");
+        if (!outputVersionMatch.Success)
+        {
+            // Unable to parse the version from the output
+            return requiredVersion;
+        }
+
+        string installedVersion = outputVersionMatch.Value;
+
+        logger.LogInformation("Installed version: {installedVersion}", installedVersion);
+
+        int comparison = CompareVersions(installedVersion, requiredVersion);
+
+        return operatorSymbol switch
+        {
+            ">=" => comparison >= 0 ? string.Empty : requiredVersion,
+            "<=" => comparison <= 0 ? string.Empty : requiredVersion,
+            ">" => comparison > 0 ? string.Empty : requiredVersion,
+            "<" => comparison < 0 ? string.Empty : requiredVersion,
+            "=" => comparison == 0 ? string.Empty : requiredVersion,
+            _ => requiredVersion,
+        };
+    }
+
+    private int CompareVersions(string version1, string version2)
```

Comment:
You could see if this might work for your case - https://learn.microsoft.com/en-us/dotnet/api/system.version.tryparse?view=net-9.0

I used it in the DotNetLanguageSpecificChecks https://github.com/Azure/azure-sdk-tools/blob/main/tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/DotNetLanguageSpecificChecks.cs#L155-L159